test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/SaveTempDataPropertyFilterTest.cs

```diff
@@ -0,0 +1,135 @@
+
+            var controller = new TestController();
+
+            var propertyHelper1 = new PropertyHelper(controller.GetType().GetTypeInfo().GetProperty("TestString"));
+            var propertyHelper2 = new PropertyHelper(controller.GetType().GetTypeInfo().GetProperty("TestString2"));
+            var propertyHelpers = new List<PropertyHelper>
+            {
+                propertyHelper1,
+                propertyHelper2
+            };
+
+            filter.PropertyHelpers = propertyHelpers;
+            var context = new ActionExecutingContext(
+                new ActionContext
+                {
+                    HttpContext = httpContext,
+                    RouteData = new RouteData(),
+                    ActionDescriptor = new ActionDescriptor()
+                },
+                new List<IFilterMetadata>(),
+                new Dictionary<string, object>(),
+                controller);
+
+            // Act
+            filter.OnActionExecuting(context);
+            controller.TestString = "SecondValue";
+            filter.OnTempDataSaving(tempData);
+
+            // Assert
+            Assert.Equal("SecondValue", controller.TestString);
+            Assert.Equal("SecondValue", tempData["TempDataProperty-TestString"]);
+            Assert.Null(controller.TestString2);
+        }
+
+        [Fact]
+        public void OnTempDataSaving_ControllerReadsTempData()
+        {
+            // Arrange
+            var httpContext = new DefaultHttpContext();
+            var tempData = new TempDataDictionary(httpContext, new NullTempDataProvider())
+            {
+                ["TempDataProperty-TestString"] = "FirstValue"
+            };
+
+            var factory = new Mock<ITempDataDictionaryFactory>();
+            factory.Setup(f => f.GetTempData(httpContext))
+                .Returns(tempData);
+
+            var filter = new SaveTempDataPropertyFilter(factory.Object);
+            var controller = new TestController();
+            var propertyHelper1 = new PropertyHelper(controller.GetType().GetTypeInfo().GetProperty("TestString"));
+            var propertyHelper2 = new PropertyHelper(controller.GetType().GetTypeInfo().GetProperty("TestString2"));
+            var propertyHelpers = new List<PropertyHelper>
+            {
+                propertyHelper1,
+                propertyHelper2
+            };
+
+            filter.PropertyHelpers = propertyHelpers;
+
+            var context = new ActionExecutingContext(
+                new ActionContext
+                {
+                    HttpContext = httpContext,
+                    RouteData = new RouteData(),
+                    ActionDescriptor = new ActionDescriptor()
+                },
+                new List<IFilterMetadata>(),
+                new Dictionary<string, object>(),
+                controller);
+
+            // Act
+            filter.OnActionExecuting(context);
+            filter.OnTempDataSaving(tempData);
```

Comment:
Yes. In the filter pipeline, it's usually called by `SaveTempDataFilter`. But for the test, it is isolated from that path. So, I need to manually make the call.